src/Raven.Server/Web/Studio/Sharding/Processors/ShardedStudioCollectionsHandlerProcessorForPreviewRevisions.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Amazon.Runtime.Internal.Endpoints.StandardLibrary;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
+using Raven.Client;
+using Raven.Client.Documents.Commands;
+using Raven.Client.Http;
+using Raven.Client.Util;
+using Raven.Server.Documents;
+using Raven.Server.Documents.Handlers.Processors.Revisions;
+using Raven.Server.Documents.Sharding.Executors;
+using Raven.Server.Documents.Sharding.Handlers;
+using Raven.Server.Documents.Sharding.Handlers.ContinuationTokens;
+using Raven.Server.Documents.Sharding.Operations;
+using Raven.Server.Documents.Sharding.Streaming;
+using Raven.Server.Documents.Sharding.Streaming.Comparers;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Web.Studio.Processors;
+using Sparrow.Json;
+
+namespace Raven.Server.Web.Studio.Sharding.Processors;
+
+internal sealed class ShardedStudioCollectionsHandlerProcessorForPreviewRevisions : AbstractStudioCollectionsHandlerProcessorForPreviewRevisions<ShardedDatabaseRequestHandler,
+    TransactionOperationContext>
+{
+    private ShardedPagingContinuation _continuationToken;
+    private string _combinedHttpEtag;
+    private CombinedReadContinuationState _combinedReadState;
+
+    public ShardedStudioCollectionsHandlerProcessorForPreviewRevisions([NotNull] ShardedDatabaseRequestHandler requestHandler) : base(requestHandler)
+    {
+    }
+
+    protected override IAsyncEnumerable<Document> GetRevisionsAsync(TransactionOperationContext context)
+    {
+        return RequestHandler.DatabaseContext.Streaming.PagedShardedStream(
+            _combinedReadState,
+            "Results",
+            ShardResultConverter.BlittableToRevisionConverter,
+            StreamDocumentByLastModifiedComparer.Instance,
+            _continuationToken).Select(s => s.Item); ;
+    }
+
+    protected override async ValueTask InitializeAsync(TransactionOperationContext context1)
+    {
+        await base.InitializeAsync(context1);
+        using (RequestHandler.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            _continuationToken = RequestHandler.ContinuationTokens.GetOrCreateContinuationToken(context);
+
+        var expectedEtag = RequestHandler.GetStringFromHeaders(Constants.Headers.IfNoneMatch);
+
+        var op = new ShardedRevisionsCollectionPreviewOperation(RequestHandler, _collection, expectedEtag, _continuationToken);
+        var result = await RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(op);
+        if (result.StatusCode != (int)HttpStatusCode.NotModified)
+            _combinedReadState = await result.Result.InitializeAsync(RequestHandler.DatabaseContext, RequestHandler.AbortRequestToken);
+        _combinedHttpEtag = result.CombinedEtag;
+    }
+
+    protected override IDisposable OpenReadTransaction(TransactionOperationContext context)
```

Comment:
because the `AbstractDatabaseHandlerProcessor` refers to it as a `TOperationContext` which is `JsonOperationContext`.
The `JsonOperationContext` does not contain the `OpenReadTransaction` method.

Reply:
but this is sharded, what do you need to read from the storage?

Reply:
Done